pkgs/by-name/qh/qhyccd_sdk/package.nix

```diff
@@ -0,0 +1,82 @@
+# This package provides the QHYCCD SDK, which is intended for non-commercial use only.
+{
+  lib,
+  stdenv,
+  fxload,
+  coreutils,
+  fetchurl,
+  makeWrapper,
+}:
+stdenv.mkDerivation rec {
+  pname = "qhyccd_sdk";
+  version = "23.09.06";
+  linux64Sha256 = "17ayxxf5wk2k14jyry3c5m5mwnyh9r0ykyh8iaxdlsk9dzjw0chp";
+  arm64Sha256 = "1ag70zpzqp6dp0rvz974vsnfyll7jvwq69s1hai3v2chvywllhs5";
+
+  dontConfigure = true;
+  dontPatch = true;
+  dontPatchELF = true;
+
+  strippedVersion = builtins.replaceStrings ["."] [""] version;
+
+  archSuffix =
+    if stdenv.system == "aarch64-linux"
+    then "Arm64"
+    else "linux64";
+
+  archSha256 =
+    if stdenv.system == "aarch64-linux"
+    then arm64Sha256
+    else linux64Sha256;
+
+  src = fetchurl {
+    url = "https://www.qhyccd.com/file/repository/publish/SDK/${strippedVersion}/sdk_${archSuffix}_${version}.tgz";
+    sha256 = archSha256;
+  };
+
+  buildInput = [makeWrapper];
+
+  buildPhase = ''
+    chmod +x ./install.sh
+    chmod +x ./distclean.sh
+    chmod +x ./uninstall.sh
+
+    mkdir -p $out/sbin
+
+    substituteInPlace ./install.sh ./distclean.sh ./uninstall.sh \
+      --replace /lib/udev $out/lib/udev \
+      --replace /etc/udev $out/etc/udev \
+      --replace "cp -a sbin/fxload" "# cp -a /sbin/fxload" \
+      --replace /usr/local $out/usr/local \
```

Comment:
This dir does not exist in Nixpkgs

Reply:
i missed the $out...will fix

Reply:
actually this is right... i'm patching the install.sh to put everything in the $out directory also commenting out two lines

Reply:
@AndersonTorres anything else is needed ?

Reply:
the sdk's `install.sh` scripts does this

```mkdir -p /usr/local/cmake_modules
cp -a usr/local/cmake_modules/*  /usr/local/cmake_modules

mkdir -p /usr/local/doc
cp -a usr/local/doc/*  /usr/local/doc

mkdir -p /usr/local/fx3load
cp -a usr/local/fx3load/*  /usr/local/fx3load

mkdir -p /usr/local/lib
cp -a usr/local/lib/*  /usr/local/lib

mkdir -p /usr/local/include
cp -a usr/local/include/*  /usr/local/include
```
as well as some directories in `/usr/share`

I can generalize the two lines
```
      --replace /usr/local $out/usr/local \
      --replace /usr/share/usb $out/usr/share/usb \
```

to one:
      --replace /usr $out/usr \

i'll do that later tonight

Reply:
changing it to a single line as above create a weird error 

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/rhb555nq1mzavnlx2fl5hdbsm5mnvl6n-sdk_linux64_23.09.06.tgz
source root is sdk_linux64_23.09.06
setting SOURCE_DATE_EPOCH to timestamp 1693982460 of file sdk_linux64_23.09.06/usr/local/fx3load/fxload
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
building
/nix/store/7cni7ndy2pm18ysl5znq6znb30sxp156-stdenv-linux/setup: line 1605: --replace: command not found

```